utils/modularize.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @author mrdoob / http://mrdoob.com/
+ */
+
+var fs = require( 'fs' );
+
+var srcFolder = '../examples/js/';
+var dstFolder = '../examples/jsm/';
+
+var files = [
+	{ path: 'controls/OrbitControls.js', ignoreList: [] },
+	{ path: 'controls/MapControls.js', ignoreList: [] },
+	{ path: 'controls/TrackballControls.js', ignoreList: [] },
+	// { file: 'controls/TransformControls.js', ignoreList: [] },
+	{ path: 'loaders/GLTFLoader.js', ignoreList: [ 'NoSide', 'Matrix2', 'DDSLoader', 'DRACOLoader', 'BufferGeometryUtils' ] },
+	{ path: 'loaders/OBJLoader.js', ignoreList: [] }
+];
+
+for ( var i = 0; i < files.length; i ++ ) {
+
+	var file = files[ i ];
+	convert( file.path, file.ignoreList );
+
+}
+
+//
+
+function convert( path, ignoreList ) {
+
+	var contents = fs.readFileSync( srcFolder + path, 'utf8' );
+
+	var className = '';
+	var dependencies = {};
+
+	// class name
+
+	contents = contents.replace( /THREE\.([a-zA-Z0-9]+) = /g, function ( match, p1 ) {
+
+		className = p1;
```

Comment:
Some files have more than one class in them. There should be a list of exports, not just one className.

Reply:
Hm, what's an example of that?

Reply:
https://github.com/mrdoob/three.js/blob/30fda3b/examples/js/controls/TransformControls.js#L622

```
THREE.TransformControls = function ( camera, domElement ) {

THREE.TransformControlsGizmo = function () {

THREE.TransformControlsPlane = function () {
```

Reply:
That's why `TransformControls` is commented out in the script 😜

Reply:
What would be preferred solution here?

a) split transform controls in multiple files/classes.

b) merge everything into one gigantic class,

Reply:
@arodic 
Other solutions have just exported multiple classes which I think is a fine approach.

```js
export { TransformControls, TransformControlsGizmo, TransformControlsPlane };
```

Reply:
```js
var classNames = [];

contents = contents.replace( /THREE\.([a-zA-Z0-9]+) = /g, function ( match, p1 ) {

		classNames.push( p1 );

		console.log( className );

		return ( classNames.length === 1 ? `_IMPORTS_\n\nvar ${p1} = ` : `var ${p1} = ` );

} );
```

```js
var exports = `export { ${classNames.join(', ')} }`;
```

Reply:
c) have ESM as the source and not worry about multiple classes ;)

Reply:
> c) have ESM as the source and not worry about multiple classes ;)

@arodic There's still a little caveat with multiple classes that is highlighted in this example https://github.com/mrdoob/three.js/pull/15518#issuecomment-451256036
when the core modules like `Scene` are both merged (not imported) into a giant `three.module.js` file which is then exported `from 'three'` and also available as separate files like `from 'three/examples/jsm/loaders/GLTFLoader'', there is a possibility that the application bundle will have two separate definitions of shared core modules. To avoid that, both `three.module.js` and `three/examples/jsm/loaders/GLTFLoader` should import from the ESM source.